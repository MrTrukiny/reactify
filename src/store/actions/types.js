export const GET_ARTISTS = 'GET_ARTISTS';
export const ARTISTS_LOADING = 'ARTISTS_LOADING';
export const GET_ALBUMS = 'GET_ALBUMS';
export const ALBUMS_LOADING = 'ALBUMS_LOADING';
export const GET_SONGS = 'GET_SONGS';
export const SONGS_LOADING = 'SONGS_LOADING';
export const PLAY_SONG = 'PLAY_SONG';
