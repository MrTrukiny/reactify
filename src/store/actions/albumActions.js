import axios from 'axios';
import { GET_ALBUMS, ALBUMS_LOADING } from '../actions/types';

export const getAlbums = id => dispatch => {
  dispatch(setAlbumsLoading());
  axios
    .get(`/artists/${id}/albums`)
      .then(res => 
        dispatch({
          type: GET_ALBUMS,
          payload: res.data.data
        })
      );
}

export const setAlbumsLoading = () => {
  return {
    type: ALBUMS_LOADING
  }
}