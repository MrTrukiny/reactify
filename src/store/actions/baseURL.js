export const baseURL = "https://rubytify.herokuapp.com/api/v1";