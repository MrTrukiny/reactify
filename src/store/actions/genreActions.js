import axios from "axios";
import { GET_GENRES, GENRES_LOADING, GET_RANDOM_SONG } from "../actions/types";

export const getGenres = id => dispatch => {
  dispatch(setGenresLoading());
  axios.get(`/genres`).then(res =>
    dispatch({
      type: GET_GENRES,
      payload: res.data.data
    })
  );
};

export const setGenresLoading = () => {
  return {
    type: GENRES_LOADING
  };
};

export const getRandomSongByGenre = genre => dispatch => {
  dispatch(setGenresLoading());
  axios.get(`/genres/${genre}/random_song`).then(res =>
    dispatch({
      type: GET_RANDOM_SONG,
      payload: res.data.data
    })
  );
};
