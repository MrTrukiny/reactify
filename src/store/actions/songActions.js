import axios from 'axios';
import { GET_SONGS, SONGS_LOADING } from '../actions/types';

export const getSongs = id => dispatch => {
  dispatch(setSongsLoading());
  axios
    .get(`/albums/${id}/songs`)
      .then(res => 
        dispatch({
          type: GET_SONGS,
          payload: res.data.data
        })
      );
}

export const setSongsLoading = () => {
  return {
    type: SONGS_LOADING
  }
}