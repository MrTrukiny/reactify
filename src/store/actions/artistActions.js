import axios from "axios";
import { GET_ARTISTS, ARTISTS_LOADING } from "../actions/types";
import { baseURL } from './baseURL'

export const getArtists = () => dispatch => {
  dispatch(setArtistsLoading());
  axios.get(`${baseURL}/artists`).then(res =>
    dispatch({
      type: GET_ARTISTS,
      payload: res.data.data
    })
  );
};

export const setArtistsLoading = () => {
  return {
    type: ARTISTS_LOADING
  };
};
