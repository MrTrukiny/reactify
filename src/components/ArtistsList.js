import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { getArtists } from "../store/actions/artistActions";
import PropTypes from "prop-types";

class ArtistsList extends Component {
  componentDidMount() {
    this.props.getArtists();
  }

  render() {
    const { artists, loading } = this.props.artists;
    return (
      <div className="container">
        {loading ? (
          <div className="container-loading">
            <h1>CARGANDO...</h1>
          </div>
        ) : (
          <div className="container-artists">
            {artists.map(artist => (
              <div className="cover-artist" key={artist.id}>
                <img src={artist.image} alt={artist.name} />
                <Link
                  to={`/artists/${artist.id}/albums`}
                  className="centered-link"
                >
                  {artist.name}
                </Link>
              </div>
            ))}
          </div>
        )}
      </div>
    );
  }
}

ArtistsList.propTypes = {
  getArtists: PropTypes.func.isRequired,
  artists: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  artists: state.artists
});

export default connect(
  mapStateToProps,
  { getArtists }
)(ArtistsList);
