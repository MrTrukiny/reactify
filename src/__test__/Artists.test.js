// Testing redux actions and rendering
import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import ArtistsList from "../components/ArtistsList";
import "../setupTest";
import * as actions from '../../src/store/actions/artistActions'
import * as types from '../../src/store/actions/types'


const mockStore = configureMockStore();
const store = mockStore({});

describe('actions', () => {
  it('should set loading to false when ARTIST are loading', () => {
    const loading = true
    const expectedAction = {
      type: types.ARTISTS_LOADING
    }
    expect(actions.setArtistsLoading(loading)).toEqual(expectedAction)
  })
})

describe("ArtistsList", () => {
  it("should render correctly", () => {
    const component = shallow(
      <Provider store={store}>
        <ArtistsList />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});