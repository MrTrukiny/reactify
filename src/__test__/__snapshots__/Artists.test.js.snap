// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsList should render correctly 1`] = `ShallowWrapper {}`;
